### PR TITLE
Sending firebase token on the login call

### LINKED ISSUE #190: Combine the login and send fcm token in one api call
Send the firebase token on the login call. There is no need for two separate calls.

Use the new login endpoint:
`POST ​/api​/v2​/access​/authorize`

Send both login info & firebase related info:
```
{
  "user": "string",
  "password": "string",
  "channelName": "string",
  "fcmToken": "string"
}
```

Remove the call to `POST ​/api​/v1​/notification​/register`

### PR DESCRIPTION
**Pull request by pedrofsn**
### What does it fix?

Closes #190 

Sending the firebase token on the login call.

### How has it been tested?

Make a login and if it has success, try to send a push. The push need to be received on device.
With this, just one call be triggered, `POST ​/api​/v2​/access​/authorize`.

### CODE REVIEW

**Comment by pedrofsn** [comment]
#hacktoberfest

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt [hunk 1]
```diff
@@ -73,35 +72,24 @@ class LoginViewModel : BaseViewModel() {
...
-        disposables.add(
-            loginRepository.registerForNotification(firebaseToken)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe({
-                    logD("Registered for notifications on firebase!")
-                    onSuccessfulRegisteredForNotification()
-                }, { throwable ->
-                    logE("Register for notification failed!", throwable)
+        loginRepository.login(user)
```
Comment (aniri) [inline app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt]:
```suggestion
        loginRepository.login(User(phone, password, firebaseToken))
```
There's no need to use a separate variable for this

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt [hunk 2]
```diff
@@ -34,10 +32,11 @@ class LoginViewModel : BaseViewModel() {
...
-    private fun onSuccessfulLogin(loginResponse: LoginResponse, firebaseToken: String) {
+    private fun onSuccessfulLogin(loginResponse: LoginResponse) {
         logD("onSuccessfulLogin")
+
         sharedPreferences.saveToken(loginResponse.accessToken)
-        registerForNotification(firebaseToken)
+        onSuccessfulRegisteredForNotification()
     }
 
     private fun onSuccessfulRegisteredForNotification() {
```
Comment (aniri) [inline app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt]: the `onSuccessfulRegisteredForNotification` method can now be merged with `onSuccessfulLogin` as there's no need for separate methods :)

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt (hunk 1 above)
Comment (pedrofsn) [reply app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt]: Done

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt (hunk 2 above)
Comment (pedrofsn) [reply app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt]: Done

**Review by aniri — APPROVED** [review]

**Pull request merged**